### PR TITLE
Date Support Added in Readability kit

### PR DESCRIPTION
**Pull request by jwd-ali**
In one of my project i was in need to add date support in your Kit. So i worked on it and finally got it working . Now i want to help others who needs date support in your library.Please review my changes. 
Thank you for your Great work.Its been great working with your kit.It will be great if you add me in contributors 
Best Regards
Jawad Ali

### CODE REVIEW

File: Example/ReadabilityKit/DetailsController.swift [hunk 1]
```diff
@@ -30,12 +30,14 @@ class DetailsController: UIViewController {
...
     @IBOutlet weak var contentTextView: UITextView?
+    @IBOutlet weak var dateText: UILabel!
     
-	var image: UIImage?
+    var image: UIImage?
 	var titleText: String?
 	var desc: String?
 	var keywords: [String]?
 	var videoURL: String?
+    var dateVlaue : String?
```
Comment (shipinev) [inline Example/ReadabilityKit/DetailsController.swift]: Need to correct typo: `dateVlaue` to `dateValue`

File: Sources/Readability.swift [hunk 2]
```diff
@@ -63,7 +64,38 @@ open class Readability {
...
+        ("//head/meta[@property='article:published_time']", "content"),
+        ("//head/meta[@name='date']", "content"),
+        ("/meta[@name='sailthru.date']", "content"),
+        ("//head/meta[@name='analyticsAttributes.articleDate']", "content"),
+        ("/meta[@property='og:article:published_time']", "content"),
+        ("//div[@class='companion-galleries embedded_story hasendslate galleries']", "data-published-date"),
+        ("//head/meta[@property='time']", "content"),
+        ("//time[@class='published-date hidden']", "datetime"),
+        ("//head/meta[@name='published_date']", "content"),
+        ("//head/meta[@property='og:title']", "content"),
```
Comment (shipinev) [inline Sources/Readability.swift]: Are you sure we need this query for date parsing?

File: Example/ReadabilityKit/DetailsController.swift (hunk 1 above)
Comment (jwd-ali) [reply Example/ReadabilityKit/DetailsController.swift]: corrected in new commit

File: Sources/Readability.swift [hunk 3]
```diff
@@ -657,4 +690,118 @@ open class Readability {
...
+            do {
+                return try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
+            } catch {
+                print(error.localizedDescription)
+            }
+        }
+        return nil
+    }
+   private func stringByRemovingControlCharacters(astring: String) -> String {
+        let controlChars = NSCharacterSet.controlCharacters
```
Comment (shipinev) [inline Sources/Readability.swift]: Is it possible to use CharacterSet struct instead of NSCharacterSet?

File: Sources/Readability.swift (hunk 2 above)
Comment (jwd-ali) [reply Sources/Readability.swift]: Yes its tested in my project 
you can test by running demo project

Comment (jwd-ali) [reply Sources/Readability.swift]: actually different sites use different date format and different tags for published date
some sites just use "published date "
some sites use "time"

File: Sources/Readability.swift [hunk 4]
```diff
@@ -657,4 +690,118 @@ open class Readability {
...
+        var range = astring.rangeOfCharacter(from: controlChars)//rangeOfCharacterFromSet(controlChars)
+        var mutable = astring
+        while let removeRange = range {
+            mutable.removeSubrange(removeRange)//removeRange(removeRange)
+            range = mutable.rangeOfCharacter(from: controlChars)//rangeOfCharacterFromSet(controlChars)
+        }
+        
+        return mutable
+    }
+    private func CheckFormat(format:String ) -> String? {
```
Comment (shipinev) [inline Sources/Readability.swift]: Could you please make the first character of the function name lower case? And for `ConvertDateToFormat` function too.

File: Sources/Readability.swift (hunk 3 above)
Comment (jwd-ali) [reply Sources/Readability.swift]: yes its possible ... changed in new commit

File: Sources/Readability.swift (hunk 2 above)
Comment (shipinev) [reply Sources/Readability.swift]: I mean this particular line `"//head/meta[@property='og:title']"` in array of queries :)

File: Sources/Readability.swift (hunk 4 above)
Comment (jwd-ali) [reply Sources/Readability.swift]: done

File: Sources/Readability.swift (hunk 2 above)
Comment (jwd-ali) [reply Sources/Readability.swift]: no ... i removed it

**Comment by shipinev** [comment]
Hey @jwd-ali. Thanks for your contribution! I will run some tests and merge your changes to the master branch after that.

**Comment by jwd-ali** [comment]
thanks.. 
please let me know if any site date is not working or you have any queries. i will try to give my best
IMG_4FD6F5E94EEC-1

**Comment by shipinev** [comment]
Everything works great. Thank you one more time :)
Currently, we don't have a contribution section in our README files. But you can always find yourself here: https://github.com/exyte/ReadabilityKit/graphs/contributors

**Comment by jwd-ali** [comment]
Always welcome sir
please recommend me as  IOS Developer if you find any related job near you :)

**Pull request merged**